### ISSUE TITLE
AdColony v4.2.0.0 and Unity 2019 LTS

### ISSUE THREAD

**Issue by JSC-Sparkling** [AdColony/AdColony-Unity-Plugin#23]
Some pieces of code don't work/build when using Unity 2019:

https://github.com/AdColony/AdColony-Unity-Plugin/blob/master/Plugin/Assets/AdColony/Editor/ADCPluginInfo.cs#L9
The minimum API level has changed to 19 for Unity 2019, anything below that was removed from the enum.

https://github.com/AdColony/AdColony-Unity-Plugin/blob/master/Plugin/Assets/AdColony/Editor/ADCPostBuildProcessor.cs#L41
`GetUnityTargetName` has been deprecated and replaced by `GetUnityFrameworkTargetGuid`.

**Comment by montree1** [comment]
Plugin/Assets/AdColony/Editor/ADCPluginInfo.cs

**Comment by montree1** [comment]
https://github.com/AdColony/AdColony-Unity-Plugin/blob/master/Plugin/Assets/AdColony/Editor/ADCPostBuildProcessor.cs

**Comment by montree1** [comment]
[]()

**Comment by montree1** [comment]
https://github.com/AdColony/AdColony-Unity-Plugin/blob/master/Plugin/Assets/AdColony/Editor/ADCPostBuildProcessor.cs

**Comment by montree1** [comment]
https://github.com/AdColony/AdColony-Unity-Plugin/issues/23

**Comment by Cookizza** [comment]
No idea on the comments on this thread but this is still an issue in Unity 2020 and it means we cannot use Adcolony.

Is there any clarity on when we could expect an update? For now we're having to migrate away from AdColony

**Comment by JSC-Sparkling** [comment]
For those wondering if there's a (temp) work-around for this issue, the answer is: yes there is.  
We've made a work-around as following:
In `Assets/AdColony/Editor/ADCPluginInfo.cs` (around line 8) we've replaced the old if-def block with a new one:
```
#if UNITY_2019_4_OR_NEWER
        public const UnityEditor.AndroidSdkVersions RequiredAndroidVersion = UnityEditor.AndroidSdkVersions.AndroidApiLevel19;
#elif UNITY_5_6_OR_NEWER
        public const UnityEditor.AndroidSdkVersions RequiredAndroidVersion = UnityEditor.AndroidSdkVersions.AndroidApiLevel16;
#else
        public const UnityEditor.AndroidSdkVersions RequiredAndroidVersion = UnityEditor.AndroidSdkVersions.AndroidApiLevel14;
#endif
```

And in `Assets/AdColony/Editor/ADCPostBuildProcessor.cs` (around line 41) we've also added an if-def block:
```
#if UNITY_2019_4_OR_NEWER
            string targetId = project.GetUnityFrameworkTargetGuid();
#else
            string targetId = project.TargetGuidByName(PBXProject.GetUnityTargetName());
#endif
```

With both of these changes applied, the plugin does compile for Unity 2019 and seems to be working just fine.

**Comment by rajeshadc** [comment]
We will update the plugin with above changes and release it soon. Thanks @JSC-Sparkling for noticing this to us.

**Comment by deamonik92** [comment]
This issue has been resolved since the release of version (4.3.2) of AdColony-Unity-Plugin
